### ISSUE TITLE
Queries using `super` datatype in Redshift are flagged as errors

### ISSUE THREAD

**Issue by puetsch** [sqlfluff/sqlfluff#1672]
```sql
SELECT
    c[0].col,
    o
FROM customer_orders c, c.c_orders o;
```

## Expected Behaviour
Query passes. This is valid syntax for querying the `super` data type (which is somewhat similar to json). https://docs.aws.amazon.com/redshift/latest/dg/query-super.html

The above query uses
- array indexing
- unnesting the array `c_orders` (which is a field of the column `customer_orders`) into rows.

## Observed Behaviour
```
L:   2 | P:   5 | L013 | Column expression without alias. Use explicit `AS` clause.
L:   2 | P:   5 | L027 | Unqualified reference 'c' found in select with more than one referenced table/view.
L:   2 | P:   9 |  PRS | Line 2, Position 9: Found unparsable section: '.col'
L:   3 | P:   5 | L027 | Unqualified reference 'o' found in select with more than one referenced table/view.
L:   4 | P:  22 | L025 | Alias 'c' is never used in SELECT statement.
L:   4 | P:  22 | L029 | Keywords should not be used as identifiers.
L:   4 | P:  36 | L025 | Alias 'o' is never used in SELECT statement.

```
## Steps to Reproduce
`sqlfluff lint --dialect=redshift -v` on above file

## Dialect
redshift

## Version
0.7.0 on Python 3.7.6

## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = postgres
templater = jinja
rules = None
exclude_rules = L009,L011,L019,L031,L034
recurse = 0
output_line_length = 160
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 120
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L007]  # Keywords
operator_new_lines = after

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = consistent

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = consistent

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = consistent

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

```

**Comment by tunetheweb** [comment]
I’d also say we’ll likely need a Redshift user to contribute - particularly for complex changes like this. That’s the way the other dialects have made most progress with someone fixing issues to solve using SQLFluff on their entire code base and massively improving that dialect support to allow that to happen.

I’ve created a wiki here to show people how dialect changes like this can be contributed if you or anyone else wish to take on that work: https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes

**Comment by PLBMR** [comment]
As of `sqlfluff==0.9.3`, all but the `L029` flag are still observed:
```bash
mkdir -p ~/.virtualenvs
python3.7 -m venv ~/.virtualenvs/issue-1672-reproduce
source ~/.virtualenvs/issue-1672-reproduce/bin/activate
pip install sqlfluff==0.9.3
...
sqlfluff lint --dialect=redshift -v super_select.sql
==== sqlfluff ====
sqlfluff:                0.9.3 python:                  3.7.4
implementation:        cpython dialect:              redshift
verbosity:                   1 templater:               jinja

==== readout ====

=== [ path: super_select.sql ] ===

== [super_select.sql] FAIL                                                                                                                                                        
L:   2 | P:   5 | L013 | Column expression without alias. Use explicit `AS` clause.
L:   2 | P:   5 | L027 | Unqualified reference 'c' found in select with more than one referenced table/view.
L:   2 | P:   9 |  PRS | Line 2, Position 9: Found unparsable section: '.col'
L:   3 | P:   5 | L027 | Unqualified reference 'o' found in select with more than one referenced table/view.
L:   4 | P:  22 | L025 | Alias 'c' is never used in SELECT statement.
L:   4 | P:  36 | L025 | Alias 'o' is never used in SELECT statement.
==== summary ====
violations:        6 status:         FAIL
All Finished 📜 🎉!
```

I will begin addressing the parsability bug (i.e. the `PRS` flag) in [this branch](https://github.com/PLBMR/sqlfluff/tree/plbmr-issue_1672_support_super_data_types).

**Comment by PLBMR** [comment]
Update: it looks like `SUPER` data types in Redshift are queried via the [PartiQL Language](https://docs.aws.amazon.com/redshift/latest/dg/query-super.html#lax-semantics). This is a SQL-compatible language for accessing relational, semi-structured, and nested data (link [here](https://partiql.org/)).

@tunetheweb In the short-term, I think it is reasonable to implement some implicit syntax of the PartiQL language in our `redshift` dialect to account for this use case. In the long-term though, I'm wondering if eventually supporting a `partiql` dialect that our `redshift` dialect partially inherits will allow us to promote adoption across non-relational DB users. `partiql` can already be used to query [DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ql-reference.html), [S3 Select](https://aws.amazon.com/blogs/opensource/announcing-partiql-one-query-language-for-all-your-data/), and [AWS Glue Data Views](https://aws.amazon.com/glue/features/).

Thoughts?

**Comment by PLBMR** [comment]
The PR above will fix all false rule violations above except for:
```bash
L:   3 | P:   5 | L027 | Unqualified reference 'o' found in select with more than one referenced table/view.
L:   4 | P:  36 | L025 | Alias 'o' is never used in SELECT statement.
```

I will address these false rule violations in a later PR.

**Comment by PLBMR** [comment]
To be clear, existing false rule violations after #2564 that were surfaced in this issue, will be addressed in issue #2567 .
